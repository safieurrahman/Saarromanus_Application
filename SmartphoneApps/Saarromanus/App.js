import React from 'react';
import { SafeAreaView } from 'react-native';

import { createAppContainer } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import { Provider } from 'react-redux';

import IndexScreen from './src/screens/index';
import SettingsScreen from './src/screens/settings';
import SightsCategoryScreen from './src/screens/sights-category';
import SightsListScreen from './src/screens/sights-list';
import SightDetailsScreen from './src/screens/sight-details';
import RouteViewScreen from './src/screens/route-view';
import RouteListScreen from './src/screens/route-list';
import LoadingScreen from './src/screens/loading';
import AlertScreen from './src/screens/alert';
import AppUseInstructionScreen from './src/screens/app-use-instruction';

import store from './src/store/';

import initOfflineSupport from './src/hooks/use-initialize-offline-support';

const navigator = createStackNavigator(
	{
		Index: IndexScreen,
		Settings: SettingsScreen,
		Sights: SightsCategoryScreen,
		SightsList: SightsListScreen,
		SightDetails: SightDetailsScreen,
		RouteView: RouteViewScreen,
		RouteList: RouteListScreen,
		AppUseInstruction: AppUseInstructionScreen,
	},
	{
		initialRouteName: 'Index',
		headerLayoutPreset: 'center',
	}
);

const App = createAppContainer(navigator);

export default function() {
	initOfflineSupport();

	return (
		<Provider store={store}>
			<SafeAreaView style={{ flex: 1 }}>
				<App data-test="App" />
				<LoadingScreen data-test="LoadingScreen" />
				<AlertScreen data-test="AlertScreen" />
			</SafeAreaView>
		</Provider>
	);
}
