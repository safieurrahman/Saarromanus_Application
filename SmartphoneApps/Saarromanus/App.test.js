import React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App from './App';

describe('App.js', () => {
	// it('demonstrates a fail test', () => {
	// 	expect(wrapper.children.length).toBe(0);
	// });
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(<App />);
	});

	it('renders correctly', () => {
		expect(wrapper).toMatchSnapshot();
		// console.log(wrapper.debug());
	});

	describe('<SafeAreaView />', () => {
		it('there should be a SafeAreaView component', () => {
			expect(wrapper.find(`SafeAreaView`).length).toBe(1);
		});
		it('<App />, <LoadingScreen /> and <AlertScreen> should be wrapped inside the SafeAreaView', () => {
			expect(wrapper.find(`SafeAreaView`).children().length).toBe(3);
		});
	});

	describe('<App />', () => {
		it('there should be an App component', () => {
			expect(wrapper.find(`[data-test='App']`).length).toBe(1);
		});
	});
	describe('<LoadingScreen />', () => {
		it('there should be a component called LoadingScreen', () => {
			expect(wrapper.find(`[data-test='LoadingScreen']`).length).toBe(1);
		});
	});
	describe('<AlertScreen />', () => {
		it('there should be a component called AlertScreen', () => {
			expect(wrapper.find(`[data-test='AlertScreen']`).length).toBe(1);
		});
	});
});
