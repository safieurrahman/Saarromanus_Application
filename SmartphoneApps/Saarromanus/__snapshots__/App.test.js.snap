// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <SafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <NavigationContainer
      data-test="App"
      theme="light"
    />
    <Connect(LoadingScreen)
      data-test="LoadingScreen"
    />
    <Connect(AlertScreen)
      data-test="AlertScreen"
    />
  </SafeAreaView>
</Provider>
`;
